client-common/src/tendermint/websocket_rpc_client.rs

```diff
@@ -126,7 +125,8 @@ impl WebsocketRpcClient {
         method: &str,
         params: &[Value],
     ) -> Result<(String, Receiver<JsonRpcResponse>)> {
-        let (message, id) = prepare_message(method, params)?;
+        let id = self.unique_id.fetch_add(1, Ordering::Relaxed).to_string();
```

Comment:
```suggestion
        let id = self.unique_id.fetch_add(1, Ordering:: SeqCst).to_string();
```

Reply:
As I understands, `Relaxed` is enough to keep it unique.

Reply:
`Relaxed` performs almost no synchronisation. In our case, this value can be accessed from multiple threads, so, it'd be good to use `Ordering::SeqCst`. You can read more about this [here](https://en.cppreference.com/w/cpp/atomic/memory_order).

Reply:
> `Relaxed` performs almost no synchronisation. In our case, this value can be accessed from multiple threads, so, it'd be good to use `Ordering::SeqCst`. You can read more about this [here](https://en.cppreference.com/w/cpp/atomic/memory_order).

We only need the atomicity of the increment operation to get the unique id, no needs to synchronize with other operations, so I think it's fine with `Relaxed` here.

Reply:
I find some explanations [here](https://vorner.github.io/2018/03/25/Atomics.html#relaxed-order--sanity-just-for-me).

Reply:
If there are two separate requests from different threads, can you explain how will this work?

Reply:
> If there are two separate requests from different threads, can you explain how will this work?

No matter what the `Ordering` is, the operation is atomic, what's what `Atomic` always guarantee.
So two concurrent request happens, either one gets 1 and the other gets 2, or one gets 2 and the other gets 1.

Reply:
Ok